### PR TITLE
Add a tokeninfo_ui_exclude policy-action

### PR DESCRIPTION
**Pull request by plettich**
Add policy to exclude tokeninfo entries in the result.

Working on #2819

### CODE REVIEW

File: privacyidea/lib/policy.py [hunk 1]
```diff
@@ -1938,7 +1939,13 @@ def get_static_policy_definitions(scope=None):
                           "attribute, set this to '*'. For more details, check "
                           "the documentation."),
                 'mainmenu': [],
-                'group': GROUP.USER}
+                'group': GROUP.USER},
+            ACTION.HIDE_TOKENINFO: {
+                'type': TYPE.STRING,
+                'desc': _('A whitespace-separated list of tokeninfo fields '
+                          'which are not displayed to the admin.'),
+                'group': GROUP.SYSTEM
```
Comment (cornelinux) [inline privacyidea/lib/policy.py:1947]: See below. Could be `GROUP.TOKEN`.

File: tests/test_api_lib_policy.py [hunk 2]
```diff
@@ -1548,6 +1549,62 @@ def test_18b_hide_audit_columns(self):
...
+        g.logged_in_user = {"username": "user1",
+                            "realm": "",
+                            "role": "user"}
+        req.all_data = {}
+        hide_tokeninfo(req)
+        self.assertNotIn('hidden_tokeninfo', req.all_data, req.all_data)
+
+        delete_policy("hide_tokeninfo_admin")
+
+        # set a policy to hide the "number" and the "realm" columns in the audit response
```
Comment (cornelinux) [inline tests/test_api_lib_policy.py:1583]: This is the old comment from the audit hide.

File: privacyidea/api/lib/prepolicy.py [hunk 3]
```diff
@@ -2127,3 +2127,29 @@ def required_piv_attestation(request, action=None):
...
+    :param action: An optional Action
+    :type action: str
+    :return: Always true. Modifies the parameter `request`
+    :rtype: bool
+    """
+    hidden_fields = Match.admin_or_user(g, action=ACTION.HIDE_TOKENINFO,
+                                        user_obj=request.User)\
+        .action_values(unique=False)
+
+    if hidden_fields:
```
[the remark this answers is not in the document]
Comment (plettich) [reply privacyidea/api/lib/prepolicy.py:2152]: will remove the `if`

File: tests/test_api_lib_policy.py (hunk 2 above)
Comment (plettich) [reply tests/test_api_lib_policy.py]: Done in 6303d623

File: privacyidea/api/lib/prepolicy.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (plettich) [reply privacyidea/api/lib/prepolicy.py]: Done in 601d8171

[the remark this answers is not in the document]
Comment (plettich) [reply privacyidea/api/lib/prepolicy.py]: BTW using a postpolicy does make things much cleaner...

File: privacyidea/lib/policy.py (hunk 1 above)
Comment (plettich) [reply privacyidea/lib/policy.py]: Done in 601d8171

File: privacyidea/lib/policy.py [hunk 4]
```diff
@@ -2068,8 +2075,13 @@ def get_static_policy_definitions(scope=None):
...
                           "For more details, check the documentation."),
                 'mainmenu': [],
-                'group': GROUP.USER}
-
+                'group': GROUP.USER},
+            ACTION.HIDE_TOKENINFO: {
+                'type': TYPE.STRING,
+                'desc': _('A whitespace-separated list of tokeninfo fields '
+                          'which are not displayed to the user.'),
+                'group': GROUP.SYSTEM
```
[the remark this answers is not in the document]
Comment (plettich) [reply privacyidea/lib/policy.py]: Done in 601d8171

**Review by cornelinux — APPROVED** [review]

**Pull request merged**